### PR TITLE
Refactor rpc so all methods async, sendRawTransaction handles arrays

### PR DESCRIPTION
**Pull request by micahriggan**

### CODE REVIEW

File: packages/bitcore-node/src/providers/chain-state/internal/internal.ts [hunk 1]
```diff
@@ -425,15 +425,13 @@ export class InternalStateProvider implements CSP.IChainStateService {
...
-      this.getRPC(chain, network).sendTransaction(rawTx, (err: any, result: any) => {
-        if (err) {
-          reject(err);
-        } else {
-          resolve(result);
-        }
-      });
-    });
+    const txids = new Array<string>();
+    const txs = typeof rawTx === 'string' ? [rawTx] : rawTx;
```
Comment (micahriggan) [inline packages/bitcore-node/src/providers/chain-state/internal/internal.ts]: rawTxs

**Comment by justinkook** [comment]
Screen Shot 2019-11-07 at 2 35 56 PM

File: packages/bitcore-node/src/providers/chain-state/internal/internal.ts [hunk 2]
```diff
@@ -425,15 +425,13 @@ export class InternalStateProvider implements CSP.IChainStateService {
...
-          reject(err);
-        } else {
-          resolve(result);
-        }
-      });
-    });
+    const txids = new Array<string>();
+    const rawTxs = typeof rawTx === 'string' ? [rawTx] : rawTx;
+    for (const tx of rawTxs) {
+      const txid = await this.getRPC(chain, network).sendTransaction(tx);
```
Comment (justinkook) [inline packages/bitcore-node/src/providers/chain-state/internal/internal.ts]: needs error handling `try catch` or `.catch`

**Comment by justinkook** [comment]
This MR fixes:
```
[warn] ERROR: 500 - "Returned error: Invalid params: invalid type: sequence, expected a 0x-prefixed, hex-encoded vector of bytes." 
```

File: packages/bitcore-node/src/providers/chain-state/internal/internal.ts (hunk 2 above)
Comment (micahriggan) [reply packages/bitcore-node/src/providers/chain-state/internal/internal.ts]: This is for bitcoin, so I actually want it to throw if it fails

**Pull request merged**